aptos-move/move-examples/feature-sandbox/sources/sandbox_messaging.move

```diff
@@ -0,0 +1,213 @@
+    ///
+    /// Private entry functions allow you to create externally callable functions in a transaction that cannot
+    /// be called from another module.  Entry functions cannot take structs as arguments, and cannot return values.
+    ///
+    /// This allows only the admin (the deployer) to override messages in the account
+    entry fun set_message_admin(
+        account: signer,
+        message_address: address,
+        message: String
+    ) acquires MessageHolder {
+        let account_addr = signer::address_of(&account);
+        check_if_admin(account_addr);
+        assert!(exists<MessageHolder<MessageChangeEvent>>(message_address), not_found(ENO_MESSAGE_HOLDER));
+        set_message_inline(message_address, message);
+    }
+
+    /// Sets a message for the signer's account
+    ///
+    /// Public entry functions allow you to create externally callable functions that can be called from other modules.
+    /// Entry functions cannot take structs as arguments, and cannot return values.
+    ///
+    /// Only the owner of the account holding the resource can update the message.
+    public entry fun set_message(account: signer, message: String)
+    acquires MessageHolder {
+        let account_addr = signer::address_of(&account);
+
+        // Create the message holder if it doesn't exist yet
+        if (!exists<MessageHolder<MessageChangeEvent>>(account_addr)) {
+            move_to(&account, MessageHolder {
+                message,
+                message_change_events: account::new_event_handle<MessageChangeEvent>(&account),
+            })
+        } else {
+            // Otherwise just update the existing message
+            set_message_inline(account_addr, message);
+        }
+    }
+
+    /// Updates an existing message in the message holder
+    ///
+    /// An inline function is inlined into the
+    inline fun set_message_inline(address: address, message: String) acquires MessageHolder {
+        let old_message_holder = get_message_holder_inline(address);
+        let old_message = *&old_message_holder.message;
+        event::emit_event(&mut old_message_holder.message_change_events, MessageChangeEvent {
+            old_message,
+            new_message: copy message,
+        });
+        old_message_holder.message = message;
+    }
+
+    /// Inline functions also can return references
+    inline fun get_message_holder_inline(address: address): &mut MessageHolder<MessageChangeEvent> acquires MessageHolder {
+         borrow_global_mut<MessageHolder<MessageChangeEvent>>(address)
+    }
+
+    #[test(account = @0x1)]
+    /// Tests can be in the module, and you can provide test addresses in the test annotation
+    ///
+    /// Each account in the test annotation must be also specified in the function signature as an argument.
+    fun sender_can_set_message(account: signer) acquires MessageHolder {
```

Comment:
nit: account can be &signer. This avoids all the extra & in the function body below